File: examples/js/foo_bar_user.mjs
export class FooBarUser {
  constructor(dependency) {
    const {foo, bar} = dependency;
    this.foo = foo;
    this.bar = bar;
  }


  danceWithDependencies() {
    const fooValue = this.foo.getSomething();
    this.bar.doSomething(fooValue);
  }
}

File: java/example/src/main/java/vanilladi/FooBarUser.java
package vanilladi;

public class FooBarUser {
    private Foo foo;
    private Bar bar;

    public FooBarUser(Foo foo, Bar bar) {
        this.foo = foo;
        this.bar = bar;
    }

    public void danceWithDependencies() {
        bar.doSomething(foo.getSomething());
    }
}
